### PR TITLE
Training script

### PR DESCRIPTION
**Pull request by evvahu**
This PQ contains a very basic training script that contains some model parts that have already been implemented.

### CODE REVIEW

**Comment by Blubberli** [comment]
I would not yet merge this because there is just too much missing (in the rest of the project). so we should first solve the dataset loading, logging and configuration. we can use this branch to work on the training script until we decide that we can merge it...

**Comment by evvahu** [comment]
second PR for this file (I won't open a new one if that's okay).

This PR includes a more elaborate version of a training script:
-loads a config file in json format from the command line as a dictionary object
-includes two methods to train a model: one for binary classification and one for multiclass classification
-includes a method for predicting on unseen data
-includes a method for getting the accuracy
-further includes a logger for logging the training process (needs to be extended)

for model.eval() and model.no_grad I've found the following:
-  model.eval() will notify all your layers that you are in eval mode, that way, batchnorm or dropout layers will work in eval mode instead of training mode.
-     torch.no_grad() impacts the autograd engine and deactivate it. It will reduce memory usage and speed up computations but you won’t be able to backprop (which you don’t want in an eval script).

I'm not sure what we want. Eval should work but other people recommend using no_grad.

**Review by Blubberli — APPROVED** [review]
Hi, I did some changes to the training so now we can merge it and I will do a new branch for model saving... 
Here are my changes:
- the multi classification with the basic classifier and static feature extractor worked well, very nice! But all other combinations did not work: 
- for binary classification the prediction of labels has to be done differently as you do not have a list of class scores and you can take the max but you have one number and the class is 0 if sigmoid(score) < 0.5, else 1. I have added a method for that in training_utils
- the binary loss expects the labels to be floats and the classifier return a tensor of size [batchsize, 1] which has to be squeezed to [batchsize]
- The training argument in the classifier has to be false in validation in the forward method (maybe this is redundant because we set the model to eval mode but I don't know how this information gets transferred to the dropout layer if we do not state it explicitly (we can maybe remove the training parameter in a new PR)
- the Bert extractor was not initialized correctly. I moved the whole dataset init to the training_utils because it is similar to the classifier init and it was quite long. the method is called get_datasets and returns the three dataset splits. There was also a bug in the bert extractor which was luckily detected by our toy test sets and resulted in NaN in the embeddings, this is fixed now and I hope that there's no other bug
- I removed the lists that store all batch losses for training and validation cause I think it is fine to just log everything in the log file. I just left the model to be returned but in general I guess we do not want to return anything as we save the best model to a checkpoint and load it from there for prediction

--> this was a complicated PR and I think we can be proud that this is done now ;)
